lib/chat/bloc/provider/ChatProvider.dart

```diff
@@ -162,13 +165,47 @@ class ChatProvider implements ViewModelProvider<ChatViewModel> {
   void _setInteractiveWidget(InputRequestEntity entity) {
     if (entity != null && entity.type != null) {
       InteractiveMessageType inputType = _getInputType(entity);
-      inputType != null
-          ? _chatViewModel.interactiveWidget = _buildInputTextWidget(
-              entity: entity,
-              inputType: inputType,
-            )
-          : _chatViewModel.interactiveWidget =
-              _buildSelectableOptionsWidget(entity: entity);
+      switch (inputType) {
+        case InteractiveMessageType.inputString:
+          _setInputTextWidget(
+            entity: entity,
+            inputType: inputType,
+          );
+          break;
+        case InteractiveMessageType.inputEmail:
+          _setInputTextWidget(
+            entity: entity,
+            inputType: inputType,
+          );
+          break;
+        case InteractiveMessageType.inputPhoneNumber:
+          _setInputTextWidget(
+            entity: entity,
+            inputType: inputType,
+          );
+          break;
+        case InteractiveMessageType.inputImage:
+          _setInputTextWidget(
+            entity: entity,
+            inputType: inputType,
+          );
+          break;
+        case InteractiveMessageType.inputDate:
+          _setDatePickerWidget(entity: entity);
+          break;
+        case InteractiveMessageType.inputDropdown:
+          _setDropdownWidget(entity: entity);
+          break;
+        case InteractiveMessageType.multiChoice:
+          _setSelectableOptionsWidget(entity: entity);
+          break;
+        default:
+          _setInputTextWidget(
+            entity: entity,
+            inputType: inputType,
+          );
+          break;
+      }
```

Comment:
We can refactor this `switch` to have something like this:

```dart
switch (inputType) {
        case InteractiveMessageType.inputDate:
          _setDatePickerWidget(entity: entity);
          break;
        case InteractiveMessageType.inputDropdown:
          _setDropdownWidget(entity: entity);
          break;
        case InteractiveMessageType.multiChoice:
          _setSelectableOptionsWidget(entity: entity);
          break;
        case InteractiveMessageType.inputString:
        case InteractiveMessageType.inputEmail:
        case InteractiveMessageType.inputPhoneNumber:
        case InteractiveMessageType.inputImage:
        default:
          _setInputTextWidget(
            entity: entity,
            inputType: inputType,
          );
          break;
      }
```
So, we don't have to repeat:

```dart
        _setInputTextWidget(
            entity: entity,
            inputType: inputType,
          );
```

Reply:
Sure, I will change it leaving only defined cases, and then the default one:
```
      switch (inputType) {
        case InteractiveMessageType.inputDate:
          _setDatePickerWidget(entity: entity);
          break;
        case InteractiveMessageType.inputDropdown:
          _setDropdownWidget(entity: entity);
          break;
        case InteractiveMessageType.multiChoice:
          _setSelectableOptionsWidget(entity: entity);
          break;
        default:
          _setInputTextWidget(
            entity: entity,
            inputType: inputType,
          );
          break;
      }
```

Reply:
Done

Reply:
this file is gonna get very big, we will address this in the refactor. we need to have registered handlers instead of a hardcoded switch for every handler.